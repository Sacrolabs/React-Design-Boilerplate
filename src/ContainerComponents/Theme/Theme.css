.theme-alignment {
  text-align: center;
  margin: 200px;
  height: 100px;
}
