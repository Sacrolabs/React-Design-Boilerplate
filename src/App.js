import React from "react";
import "./App.css";
import { Switch,Route } from "react-router-dom";
import TodoList from "./ContainerComponents/Todo/TodoList";
import Theme from "./ContainerComponents/Theme/Theme";
import Footer from "./ContainerComponents/Layout/Footer/Footer";
import HomePage from "./ContainerComponents/HomePage/HomePage";
import GitUserDetails from "./ContainerComponents/GitUserDetails/GitUserDetails";
import Header from './ContainerComponents/Layout/Header/Header';
const App = () => {
  return (
    <>
    <Header/>
      <Switch>
        <Route path="/Home" component={HomePage} />
        <Route path="/UserInfoGit" component={GitUserDetails} />
        <Route path="/TodoList" component={TodoList} />
        <Route path="/Theme" component={Theme} />
      </Switch>
      <Footer/>
    </>
  );
};

export default App;