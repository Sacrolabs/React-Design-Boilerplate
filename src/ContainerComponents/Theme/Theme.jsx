import React, { useState } from "react";
import "./Theme.css";
const Theme = () => {
  const [theme, setTheme] = useState("light");

  const toggleTheme = () => {
    setIsActive((current) => !current);
    const nextTheme = theme === "light" ? "dark" : "light";
    setTheme(nextTheme);
  };
  const [isActive, setIsActive] = useState(false);

  return (
    <div className="theme-alignment">
      <button
        style={{
          backgroundColor: isActive ? "black" : "",
          color: isActive ? "white" : "",
        }}
        onClick={toggleTheme}
      >
        Current theme: {theme}
      </button>
    </div>
  );
};

export default Theme;
