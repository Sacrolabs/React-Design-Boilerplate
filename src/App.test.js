import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import { Provider } from "react-redux";
import { configureStore } from "./store";
import { BrowserRouter } from "react-router-dom";
import Theme from "./ContainerComponents/Theme/Theme";

test("Test theme button toggle", () => {
  render(<Theme />);
  const buttonEl = screen.getByText(/Current theme/i);

  userEvent.click(buttonEl);
  expect(buttonEl).toHaveTextContent(/dark/i);
});

it("should take a snapshot", () => {
  const { asFragment } = render(
    <Provider store={configureStore()}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );

  expect(
    asFragment(
      <Provider store={configureStore()}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    )
  ).toMatchSnapshot();
});
