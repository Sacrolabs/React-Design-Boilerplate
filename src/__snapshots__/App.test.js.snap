// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="header-top__navigation"
  >
    <section
      class="navbar"
    >
      <a
        class="navbar-item"
        href="/Home"
      >
        Home
      </a>
      <a
        class="navbar-item"
        href="/UserInfoGit"
      >
        User-Info
      </a>
      <a
        class="navbar-item"
        href="/TodoList"
      >
        Add Todo
      </a>
      <a
        class="navbar-item"
        href="/Theme"
      >
        Theme
      </a>
      <a
        class="navbar-item"
        href="/About"
      >
        About
      </a>
    </section>
  </section>
  <main
    class="footer-main"
  >
    <p>
       Â© 2022 SacroLabs Softwares, All rights reserved. 
    </p>
  </main>
</DocumentFragment>
`;
